### ISSUE TITLE
Installer does not support upgrade?

### ISSUE THREAD

**Issue by jol64** [cedrozor/myrtille#127]
when I tried to install Myrtille 2.0 I got the following dialog:
![image](https://user-images.githubusercontent.com/24666601/43999690-e50fe86a-9e11-11e8-8eca-a229e6dce9c6.png)
Can you please support direct upgrades?
I would also prefer to download and run just a .msi. Prerequesite checks or installations should be possible as prerequisite declarations (.Net, nowadays usually installed already) or custom actions (e.g. installating a Windows feature).

**Comment by cedrozor** [comment]
Yes, I'm aware of that inconvenience. I tried to have the installer in automatic upgrade mode but I encountered problems with files not being replaced (i.e. config files, if previously modified by the user) or removed (if no longer exist in newer version), so I preferred a clean uninstall before installing a new version.

So I enabled"**RemovePreviousVersions**" into the installer properties, but with no effect. It works well under Visual Studio (right-click on the setup project and choose "Install"), correctly uninstalling Myrtille first if it's already installed (then proceeding to the new install), but displaying the dialog box you posted otherwise.

![image](https://user-images.githubusercontent.com/17749498/44003530-16530212-9e54-11e8-9bf0-b8c91804d740.png)

I'm not the first to have such issue: https://stackoverflow.com/questions/8640475/visual-studio-set-up-doesnt-uninstall-previous-version-of-the-build

I tried to change the **ProductCode** and **UpgradeCode** but without success (it doesn't uninstall a previous installation first). The only thing that works safely is to change the **PackageCode** (directly into the .vdproj file), but it displays the dialog box and the user have to uninstall first manually.

I probably missing something but I don't know what exactly. If someone have an idea, I'll take it!

Anyway, I think the VS Installer project extension (https://marketplace.visualstudio.com/items?itemName=VisualStudioClient.MicrosoftVisualStudio2017InstallerProjects) is deprecated and quite limited. I should replace it, there are many tools for that: Wix, Installshield, etc.

Regarding the .msi, you can already run it directly and bypass the bootstrapper (setup.exe); whose role is to check and download the myrtille prerequisistes, if needed. .NET isn't always installed already and there may be others (for example, previouses versions of Myrtille required the VC++ redist).

**Comment by jol64** [comment]
I am using wix. At least allows to specify .Net requirements (and starting with Win 8 there is always one .Net version preinstalled and then updated, c.f. https://docs.microsoft.com/de-de/dotnet/framework/migration-guide/versions-and-dependencies). Windows features can be tested for with WMI and installed with dism – if you need code examples please let me know.

Upgrading anything a user might modify is always a nightmare unless you use version control like git. An imho decent example of how to do it reasonably is mailcow-dockerized which uses git to version all configuration.

**Comment by cedrozor** [comment]
I'm using the VS Installer project extension for historic reasons, and because the customs actions were written for the MSI engine. But I think Wix is better, and the transition should be quite easy (basically importing the myrtille custom actions) because it's also based on MSI.

Sadly, I don't have much time for such thing :/ would you like to do it?
